### PR TITLE
Bug fixes in dataframe 

### LINKED ISSUE #415: add SortOptions to df.sort_values(...)

### LINKED ISSUE #416: df.concat uses Table.union instead of Table.concat
This Table.Union would remove duplicates, whereas the concat semantic is to keep duplicates

### PR DESCRIPTION
**Pull request by nirandaperera**
fixes #415 add SortOptions to df.sort_values(...)
fixes #416 df.concat uses Table.union instead of Table.concat

### CODE REVIEW

File: python/examples/dataframe/concat.py [hunk 1]
```diff
@@ -1,17 +1,28 @@
...
 from pycylon.net import MPIConfig
-import random
 
-df1 = DataFrame([random.sample(range(10, 100), 50),
-                 random.sample(range(10, 100), 50)])
-df2 = DataFrame([random.sample(range(10, 100), 50),
-                 random.sample(range(10, 100), 50)])
-df3 = DataFrame([random.sample(range(10, 100), 50),
-                 random.sample(range(10, 100), 50)])
+t1 = pa.table([random.sample(range(10, 100), 5),
```
Comment (vibhatha) [inline python/examples/dataframe/concat.py:8]: Is this necessary?

File: python/pycylon/data/table.pyx [hunk 2]
```diff
@@ -2366,7 +2365,7 @@ cdef class Table:
         return PyLocIndexer(self, "iloc")
 
     @staticmethod
-    def concat(tables: List[Table], axis: int = 0, join: str = 'inner', algorithm: str = 'sort', distributed=False):
+    def concat(tables: List[Table], axis: int = 0, join: str = 'inner', algorithm: str = 'sort'):
```
Comment (vibhatha) [inline python/pycylon/data/table.pyx:2368]: A local concat operation is needed even in the distributed setting. 
Shall we add separate methods for local and distributed or we can keep this flag?

Comment (vibhatha) [reply python/pycylon/data/table.pyx:2368]: Note: This is used as it is in the UNO application

Comment (nirandaperera) [reply python/pycylon/data/table.pyx:2368]: Oh. I see your point. Ishall we add a `distributed_concat` to the table API (like in other operations)?

Comment (vibhatha) [reply python/pycylon/data/table.pyx:2368]: Yeah that sounds great :+1:

Comment (nirandaperera) [reply python/pycylon/data/table.pyx:2368]: great. I'll add that now. And the use it in dataframe api accordingly.

File: python/pycylon/frame.py [hunk 3]
```diff
@@ -1858,6 +1689,7 @@ def sort_values(
             na_position="last",
             ignore_index=False,
             key=None,
+            sort_options: Dict = None,
```
Comment (chathurawidanage) [inline python/pycylon/frame.py:1692]: @supunkamburugamuve

Comment (chathurawidanage) [inline python/pycylon/frame.py:1692]: @nirandaperera Let's change this to accept multiple args instead.

**Review by vibhatha — APPROVED** [review]

**Review by chathurawidanage — APPROVED** [review]

**Pull request merged**